packages/compiler/core/program.ts

```diff
@@ -899,7 +906,19 @@ export async function compile(
     const [pkg] = await loadFile(host, pkgJsonPath, JSON.parse, program.reportDiagnostic, {
       allowFileNotFound: true,
     });
-    const mainFile = resolvePath(dir, typeof pkg?.tspMain === "string" ? pkg.tspMain : "main.tsp");
+    // Back Compat: if main.cadl exist, return main.cadl
+    let mainFile = resolvePath(dir, "main.cadl");
```

Comment:
It seems inconsistent to give main.cadl priority here.

I think the logic needs to be something like this:

```
// 1. Use tspMain (with fallback to cadlMain) 
const tspMain = getTspMain(pkg);
if (tspMain) {
 return tspMain;
}

// 2. Use main.cadl if it exists
/* ...(Put this back compat code here)... */

// 3. Use main.tsp
return "main.tsp";